[PyTorch][Kineto] add ActivityType.h when USE_KINETO is not set

shintaro-iwasaki opened the pull request:
Summary:
This patch fixes an error "'ActivityType.h' file not found" when `use_kineto()` is false.

## Problem
Even when `use_kineto()` is not set (i.e., `-DUSE_KINETO` is not passed), `ActivityType.h` is required for PyTorch compilation:
https://github.com/pytorch/pytorch/blob/master/torch/csrc/profiler/kineto_shim.h#L15

## Solution
Add `ActivitiyType.h` dependency even when `use_kineto() == False`.

Test Plan: PyTorch internal and external CI tests.

Differential Revision: D38090153

kit1980 commented:
It's strange to require a file from kineto with use_kineto() == False.
A better fix would be to modify the code so it doesn't need ActivityType.h (a file from kineto) when kineto is not used...

shintaro-iwasaki commented:
This dependency is because of this PR: #81450
@robieta: I guess there's a context behind this: may I ask why PyTorch now needs `ActivityType.h` even when Kineto is not used?

robieta commented:
The reason for the change is that we want to have access to constants from Kineto (e.g. enums) for our own types, even if we aren't actually using Kineto. (This PR should add `//kineto/libkineto:libkineto_headers`) It saves a lot of `#ifdef` and forward declaration headache to just be able to use ActivityType directly.

shintaro-iwasaki commented:
Thanks for your feedback! I personally prefer the current design (`//kineto/libkineto:kineto_activity_header`) because:
1. It minimizes the dependency by limiting the exposure of headers (now only `ActivityType.h`).
2. It looks more temporary. I feel this Kineto-PyTorch dependency should be fundamentally solved by either
   - Fully integrating Kineto in PyTorch; or
   - Cleaning up the interface so that PyTorch does not need to rely on Kineto's header when Kineto is disabled as suggested by @kit1980.

I would love to hear your feedback: I believe both of you (@robieta and @kit1980) know more about the context of PyTorch and Kineto.

robieta commented:
> 1. It minimizes the dependency by limiting the exposure of headers (now only `ActivityType.h`).

That's an excellent point. And of course I'm open to how we can better manage the dependency.

shintaro-iwasaki committed:
[PyTorch][Kineto] add ActivityType.h when USE_KINETO is not set (#82028)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/82028

This patch fixes an error "'ActivityType.h' file not found" when `use_kineto()` is false.

## Problem
Even when `use_kineto()` is not set (i.e., `-DUSE_KINETO` is not passed), `ActivityType.h` is required for PyTorch compilation:
https://github.com/pytorch/pytorch/blob/master/torch/csrc/profiler/kineto_shim.h#L15

## Solution
Add `ActivitiyType.h` dependency even when `use_kineto() == False`.

Test Plan: PyTorch internal and external CI tests.

Differential Revision: D38090153

fbshipit-source-id: 8a1dda0165dcbe61b00894a702906fbed587d6c2

shintaro-iwasaki commented:
Would you review this PR? @kit1980 or @robieta?

It seems that others also faced this issue (see Diff), and this patch should unblock us.  Particularly, it will affect neither OSS PyTorch build (because the build system is different) nor our internal standard packages (because Kineto is enabled by default).

kit1980 commented:
> Would you review this PR? @kit1980 or @robieta?
> 
> It seems that others also faced this issue (see Diff), and this patch should unblock us. Particularly, it will affect neither OSS PyTorch build (because the build system is different) nor our internal standard packages (because Kineto is enabled by default).

LGTM

robieta reviewed (approved):
LGTM. Thanks for not only the fix but also the thoughtful analysis that accompanied it.

shintaro-iwasaki commented:
Thank you very much for you prompt reviews, @kit1980 or @robieta!
